daemons/gptp/common/ptp_message.cpp

```diff
@@ -1364,18 +1366,19 @@ void PTPMessagePathDelayRespFollowUp::processMessage(IEEE1588Port * port)
 	 * than a certain number of lost FUPs we should set asCapable
 	 * to false
 	 */
+	diffLastSeqId = sequenceId - port->getLastSeqId();
```

Comment:
Keep in mind that the "usual arithmetic conversions" apply to all the basic math operators. And the first step of those conversions, when dealing with integral operands, is to apply the "integer promotions".

The integer promotions will convert any integral type of *lesser rank* than the rank of `int` and `unsigned int` to be of type `int` so long as all the values of the original type will fit in `int`. Otherwise the type will be promoted to `unsigned int`.  Rank generally corresponds to the bit-width of the type, with wider types having higher rank, and the signed/unsigned versions having the same rank. So in this case, the `uint16_t` values are converted to `int`, since all `uint16_t` values fit in the range of `int`.

So the result of subtracting two `uint16_t` values is a value of type `int`.  In a wrap-around situation, it will be a small negative number.  Now, when that value is assigned to a variable, the value on the right hand side of the assignment expression takes on the type of the left-hand side.

In the case of the left-hand side having type `unsigned int`, the negative number can't be represented in that type. Therefore, the unsigned conversion rule applies, which repeatedly adds or subtracts one more than the maximum representable value of the target type to the value until the result falls in the representable range. In this example, we add `UINT_MAX+1` to `-1` and end up with `UINT_MAX`. Fortunately, C provides reasonable semantics for unsigned conversion and overflow; in the case of values that don't fit into signed types (or which overflow), the result is undefined.